llvm/lib/Target/RISCV/RISCVAsmPrinter.cpp

```diff
@@ -152,8 +153,40 @@ void RISCVAsmPrinter::LowerPATCHPOINT(MCStreamer &OutStreamer, StackMaps &SM,
 
   PatchPointOpers Opers(&MI);
 
+  const MachineOperand &CalleeMO = Opers.getCallTarget();
   unsigned EncodedBytes = 0;
 
+  if (CalleeMO.isImm()) {
+    uint64_t CallTarget = CalleeMO.getImm();
+    if (CallTarget) {
+      assert((CallTarget & 0xFFFF'FFFF'FFFF) == CallTarget &&
+             "High 16 bits of call target should be zero.");
+      // Materialize the jump address:
+      SmallVector<MCInst, 8> Seq;
+      RISCVMatInt::generateMCInstSeq(CallTarget, *STI, RISCV::X1, Seq);
+      for (MCInst &Inst : Seq) {
+        MCInst CInst;
+        bool compressed = RISCVRVC::compress(CInst, Inst, *STI);
+        EmitToStreamer(OutStreamer, Inst);
```

Comment:
```suggestion
        EmitToStreamer(OutStreamer, Compressed ? CInst : Inst));
```

Reply:
The reason I used `Inst` instead of `CInst` was that `EmitToStreamer` increments `RISCVNumInstrsCompressed` if the instruction is compressed. In this case, should I use CInst and increment `RISCVNumInstrsCompressed` here if needed?

Reply:
I have created a PR (#88120) so that we don't have to call `RISCVRVC::compress` twice.
And after that, you can write:
```
bool Compressed = EmitToStreamer(OutStreamer, Inst);
EncodedBytes += Compressed ? 2 : 4;
```